### DISCUSSION TITLE
Description field in favorites, routes, tracks

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#346] [Q&A] [answered]
Hello everybody! Maybe I missed something, but why haven't we thought about short comments (descriptions) for our Favorites, Tracks or Routes yet? 🙂

I find this extremely useful. This allows you to add a opening hours, phone number, address or name of a person you know mainly by nickname to your Favorite. I see lots of uses for such comments added to Tracks or Routes too, especially if it were included in the global search.

From what I know, a GPX file can contain `<cmt>` and `<desc>` tags for this purpose. Maybe there is a better way to do this? I'm sure @0709wiwiwi knows more about GPX than I do 🙂

**Comment by devemux86** [comment]
@Ciutnik very useful idea.

We could use more fields in the favorites / route / tracks.

For reference see the available fields in GPX `wpt`:
https://www.topografix.com/GPX/1/1/#type_wptType

**Reply by Ciutnik** [reply]
In fact, e.g. `<link>` can also be useful.

**Reply by 0709wiwiwi** [reply]
The gpx 1.1 standard recommends:

Element: cmt 
GPS waypoint comment. Sent to GPS as comment.

Element: desc 
A text description of the element. 
Holds additional information about the element intended for the user, not the GPS.

Cruiser current uses:
In navtrack trackpoints (gpx trkpt).
cmt: Contains the street name.
desc: (spare)

In the individual waypoints (gpx wpt)
cmt: (spare)
desc: Put free additional (text) info.

Cruiser has an extension in waypoints export.
Adding extra elements should be possible.
Desired: A KISS offer ;-) 

[Cruiser_favorites_x.gpx.txt](https://github.com/devemux86/cruiser/files/12916784/Cruiser_favorites_x.gpx.txt)

![image](https://github.com/devemux86/cruiser/assets/54167201/2065b207-6ad1-4e56-8620-a4407023f18f)

**Reply by ramondevic** [reply]
Maybe only with `<desc>` would be enough so that everyone can use it to their liking

(original in Catalan)

**Reply by devemux86** [reply]
Description seems a simple first step for all 3 types: favorites, routes, tracks.

I do not know if we need more fields?
Only for favorites or for all 3 types?

**Reply by Ciutnik** [reply]
It seems to me that `<desc>` (and only that) is useful for all three of our data types.

**Reply by 0709wiwiwi** [reply]
Add the gpx desc in as well wpt & rte & trk would sure be nice and usefull.  
As you can add (anno Y2023) rich text (inc follow links into the description.

**Comment by MotoUKRider** [comment]
For info: Just checked with BaseCamp and I can output the "Notes" field from a Route inside `<rte>` to my usual GPX that I load into Cruiser:

`<desc>A trip around the Yorkshire Dales</desc>`

When I select all the Waypoints and export them to a single GPX (to display in Cruiser), I get 2 entries within the `<wpt>` tag:
`<cmt>Notes field is output here</cmt>`
`<desc>Notes field is output here</desc>`

**Reply by 0709wiwiwi** [reply]
@MotoUKRider 

Garmin? Ok let's talk about Garmin, a little, and only once. 
Garmin does not use descripition but notes, strange right ? 
Garmin publishes identical info in both cmt and desc which is redundant.
Garmin has either not  studied or not understood the gpx topograxics recomandations ? 

The gpx standard recommends it this way.
In **cmt** you put **compact** msg's which the gps can show in the navigation screen.  
In **desc** you can place **extra** info that is available to the user after calling.

This recommendation is 100% respected in the Locus gpx navigation track.
A navigation trackpoint is composed as follows.
![image](https://github.com/devemux86/cruiser/assets/54167201/32c8875f-6a16-4d9a-8779-08206ea699eb)
In a gpx navtrkpt the street name is attached by the router into the cmt.
You show the (cmt) or street name so in the navigation display.
Additional information can still be placed into the navtrktpt desc. 
This way desc info does not unnecessarily clutter the navigation screen.
The user has to call up that extra desc info separately.

Proposal with discrete use of desc in via points is to come *soon.
Sorry for possible typos as this is a holiday edit.

* Via Points can have name and desc.
https://github.com/devemux86/cruiser/discussions/401

**Comment by ramondevic** [comment]
One of the waypoint files I use very often has the fields `<ele><name><cmt><sym><type>`

Cruiser only takes `<name>` and I wish it could also use `<desc>` and `<type>` (I don't think anything else is needed) and be able to choose if `<desc>` is displayed or not... although if if you could filter by `<type>` it would be great ;)


```
<wpt lat="42.26903" lon="1.58607">
    <ele>1780.0</ele>
    <name>quiet spot in the mountains</name>
    <cmt>It&#39;s alongside a trail, very quiet and beautiful surroundings
Open: Yes
Electricity: No
Wifi: No
Kitchen: No
Restaurant: No
Showers: No
Water: No
Toilets: No
Big Rig Friendly: Yes
Tent Friendly: Yes
Pet Friendly: Yes
</cmt>
    <desc>It&#39;s alongside a trail, very quiet and beautiful surroundings</desc>
    <sym>Contact, Pirate</sym>
    <type>Wild Camping</type>
</wpt>
```


![foto1](https://github.com/devemux86/cruiser/assets/122691401/c143decd-2cca-46ce-91c4-e026e0f96ab6)
![foto2](https://github.com/devemux86/cruiser/assets/122691401/ead47253-df3d-45e7-b770-6d3bd8389449)

**Comment by devemux86** [comment]
**Description** field in favorites, routes and tracks will be available as part of the new:
- https://github.com/devemux86/cruiser/discussions/706

**Comment by devemux86** [comment]
Description field in favorites, routes, tracks is available in:
- https://github.com/devemux86/cruiser/discussions/734

**Comment by devemux86** [comment, marked as answer]
- Description field in favorites, routes, tracks is available in https://github.com/devemux86/cruiser/discussions/757
